### PR TITLE
Prompt for nic for external access & configure it

### PR DESCRIPTION
**Pull request by gnuoy**
If the user requests remote access then ask a follow up question asking them to provide a nic to be used for north south traffick. Pass the nic name to the openstack-hypervior for it to wire it into the bridge.

### CODE REVIEW

**Comment by gnuoy** [comment]
Corresponding hypervisor PR https://github.com/openstack-snaps/snap-openstack-hypervisor/pull/21

File: sunbeam/commands/configure.py [hunk 1]
```diff
@@ -39,7 +41,27 @@
...
-snap = Snap()
+
+
+def get_free_nic() -> str:
+    """Return a nic which does not have a v4 or v6 address."""
+    virtual_nic_dir = "/sys/devices/virtual/net/"
+    virtual_nics = [p.name for p in Path(virtual_nic_dir).glob("*")]
+    free_nics = []
+    for nic_name, nic_config in psutil.net_if_addrs().items():
+        if nic_name in virtual_nics:
```
Comment (javacruft) [inline sunbeam/commands/configure.py:52]: Does a bond interface show up as a virtual network interface?

File: sunbeam/ohv_config/config.py [hunk 2]
```diff
@@ -86,6 +86,7 @@ class NetworkConfig(BaseModel):
     ovn_cacert: Optional[str] = Field(alias="ovn-cacert")
     enable_gateway: bool = Field(alias="enable-gateway", default=False)
     ip_address: Optional[IPvAnyAddress] = Field(alias="ip-address")
+    gateway_nic: Optional[str] = Field(alias="gateway-nic")
```
Comment (javacruft) [inline sunbeam/ohv_config/config.py:89]: I'm not convinced that the label 'gateway_nic' makes sense to me - generally we've used the term 'external' across the charms to describe this piece of connectivity - might be worth a discussion more generally about terminology.

File: requirements.txt [hunk 3]
```diff
@@ -14,6 +14,7 @@ semver # BSD-3
 
 # Used for getting local ip address
 netifaces
+psutil
```
Comment (javacruft) [inline requirements.txt:17]: netifaces already has quite alot of API for interrogating the interfaces on a machine - do we need psutil as well?

**Review by javacruft — APPROVED** [review]

**Pull request merged**